gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/RaoValidator.java

```diff
@@ -57,7 +66,13 @@ public DichotomyStepResult<RaoResponse> validateNetwork(Network network, Dichoto
             RaoResponse raoResponse = raoRunnerClient.runRao(raoRequest);
             LOGGER.info("[{}] : RAO response received: {}", direction, raoResponse);
             RaoResult raoResult = fileImporter.importRaoResult(raoResponse.getRaoResultFileUrl(), fileImporter.importCracFromJson(raoResponse.getCracFileUrl()));
-            return DichotomyStepResult.fromNetworkValidationResult(raoResult, raoResponse);
+            if (isPortugalInDirection() && raoResultIsSecure(raoResult)) {
+                AngleMonitoring angleMonitoring = new AngleMonitoring(sweData.getCracEsPt().getCrac(), network, raoResult, fileImporter.importCimGlskDocument(sweData.getGlskUrl()));
+                AngleMonitoringResult angleMonitoringResult = angleMonitoring.run(LoadFlow.find().getName(), LoadFlowParameters.load(), 4, sweData.getTimestamp());
+                businessLogger.info("[{}] : Angle monitoring result {}", direction, angleMonitoringResult.getStatus());
+                return DichotomyStepResult.fromNetworkValidationResult(raoResult, new SweDichotomyValidationData(raoResponse, angleMonitoringResult), angleMonitoringResult.isSecure());
+            }
+            return DichotomyStepResult.fromNetworkValidationResult(raoResult, new SweDichotomyValidationData(raoResponse, null));
```

Comment:
Can you create a second constructor of SweDichotomyValidationData taking only one argument ? Like :
public SweDichotomyValidationData(Rao raoresponse) {
return new SweDichotomyValidationData(raoresponse, null)
}
TBD

Reply:
added new constructor and changed call to use new cosntructor if angloe monitoring check is null